lib/class-wp-rest-url-details-controller.php

```diff
@@ -208,13 +214,92 @@ private function get_remote_url( $url ) {
 	 * @return string the title tag contents (maybe empty).
 	 */
 	private function get_title( $html ) {
-		preg_match( '|<title>([^<]*?)</title>|is', $html, $match_title );
+		preg_match( '|<\s*title[^>]*>(.*?)<\s*/\s*title>|is', $html, $match_title );
 
-		$title = isset( $match_title[1] ) ? trim( $match_title[1] ) : '';
+		$title = isset( $match_title[1] ) && is_string( $match_title[1] ) ? trim( $match_title[1] ) : '';
 
 		return $title;
 	}
 
+	/**
+	 * Parses the site icon from the provided HTML
+	 *
+	 * @param string $html the HTML from the remote website at URL.
+	 * @param string $url the target website URL.
+	 * @return string the icon URI (maybe empty).
+	 */
+	private function get_icon( $html, $url ) {
+		preg_match( '|<link.*?rel="\s*[shortcut]+(?:\s+[icon]+)*\s*".*?href="(.*?)".*?\/?>|is', $html, $matches );
+
+		$icon = isset( $matches[1] ) && is_string( $matches[1] ) ? trim( $matches[1] ) : '';
+
+		// Attempt to convert relative URLs to absolute.
+		if ( ! empty( $icon ) ) {
+			$parsed = parse_url( $icon );
+			if ( empty( $parsed['host'] ) ) {
+				$icon = $url . $icon;
+			}
+		}
+
+		return $icon;
+	}
+
+	/**
+	 * Parses the meta description from the provided HTML.
+	 *
+	 * @param string $html the HTML from the remote website at URL.
+	 * @return string the meta description contents (maybe empty).
+	 */
+	private function get_description( $html ) {
+		$description = '';
+
+		$temp = tmpfile();
+
+		if ( ! $temp ) {
+			fclose( $temp ); // clean up tmp file.
+			return $description;
+		}
+
+		$path = stream_get_meta_data( $temp )['uri'];
+
+		// Write HTML.
+		fwrite( $temp, $html );
+
+		$meta = get_meta_tags( $path );
+
+		if ( empty( $meta ) ) {
+			fclose( $temp ); // clean up tmp file.
+			return $description;
+		}
+
+		$description = ! empty( $meta['description'] ) ? $meta['description'] : '';
+
+		fclose( $temp ); // clean up tmp file.
+		return $description;
+	}
+
+	/**
+	 * Parses the Open Graph Image from the provided HTML.
+	 *
+	 * See: https://ogp.me/.
+	 *
+	 * @param string $html the HTML from the remote website at URL.
+	 * @param string $url the target website URL.
+	 * @return string the OG image (maybe empty).
+	 */
+	private function get_image( $html, $url ) {
+		preg_match( '|<meta.*?property="og:image[:url]*?".*?content="(.*?)".*?\/?>|is', $html, $matches );
```

Comment:
@hellofromtonya Here we are only interested in

`property="og:image"`

or

`property="og:image:url`

Any other variations should be ignored.